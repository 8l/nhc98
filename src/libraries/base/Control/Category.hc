#include "newmacros.h"
#include "runtime.h"

#define CT_v283	((void*)startLabel+32)
#define CT_v284	((void*)startLabel+68)
#define CT_v285	((void*)startLabel+108)
#define CT_v286	((void*)startLabel+144)
#define CT_v287	((void*)startLabel+196)
#define FN_LAMBDA280	((void*)startLabel+224)
#define CT_v289	((void*)startLabel+244)
#define CF_LAMBDA280	((void*)startLabel+252)
#define CT_v290	((void*)startLabel+280)
#define FN_LAMBDA281	((void*)startLabel+308)
#define CT_v292	((void*)startLabel+328)
#define CF_LAMBDA281	((void*)startLabel+336)
#define CT_v293	((void*)startLabel+360)
#define CT_v294	((void*)startLabel+392)
#define CT_v295	((void*)startLabel+432)
#define ST_v288	((void*)startLabel+452)
#define ST_v291	((void*)startLabel+493)
extern Node FN_Control_46Category_46_46[];
extern Node FN_Prelude_46_46[];
extern Node F0_Prelude_46id[];
extern Node FN_NHC_46Internal_46_95noMethodError[];

static Node startLabel[] = {
  bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v283)
,};
Node FN_Control_46Category_46_62_62_62[] = {
  bytes2word(NEEDHEAP_I32,PUSH_ARG_I2,PUSH_ARG_I3,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I3,HEAP_ARG,1,ZAP_ARG_I1)
, bytes2word(ZAP_ARG_I2,ZAP_ARG_I3,EVAL,NEEDHEAP_I32)
, bytes2word(APPLY,2,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v283: (byte 0) */
  HW(1,3)
, 0
,};
Node F0_Control_46Category_46_62_62_62[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_62_62_62),3)
, VAPTAG(useLabel(FN_Control_46Category_46_46))
, bytes2word(1,0,0,1)
, useLabel(CT_v284)
,};
Node FN_Control_46Category_46_60_60_60[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_ARG)
, bytes2word(1,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v284: (byte 0) */
  HW(1,1)
, 0
,};
Node F0_Control_46Category_46_60_60_60[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_60_60_60),1)
, VAPTAG(useLabel(FN_Control_46Category_46_46))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v285)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62_46_46[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_ARG_ARG_RET_EVAL)
, bytes2word(1,2,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v285: (byte 0) */
  HW(1,2)
, 0
,};
Node F0_Control_46Category_46Category_46Prelude_46_45_62_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62_46_46),2)
, VAPTAG(useLabel(FN_Prelude_46_46))
, bytes2word(0,0,0,0)
, useLabel(CT_v286)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62_46id[] = {
  bytes2word(NEEDSTACK_I16,PUSH_CVAL_P1,3,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v286: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_Control_46Category_46Category_46Prelude_46_45_62_46id[] = {
  VAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62_46id))
, useLabel(F0_Prelude_46id)
, bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v287)
,};
Node FN_Control_46Category_46_95_46_46[] = {
  bytes2word(ZAP_ARG_I1,NEEDHEAP_I32,PUSH_ARG_I3,PUSH_ARG_I2)
, bytes2word(HEAP_CVAL_I3,PUSH_HEAP,HEAP_CVAL_I4,HEAP_OFF_N1)
, bytes2word(2,ZAP_ARG_I2,ZAP_ARG_I3,EVAL)
, bytes2word(NEEDHEAP_I32,APPLY,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v287: (byte 0) */
  HW(2,3)
, 0
,};
Node F0_Control_46Category_46_95_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_95_46_46),3)
, VAPTAG(useLabel(FN_LAMBDA280))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95noMethodError))
, bytes2word(0,0,0,0)
, useLabel(CT_v289)
,	/* FN_LAMBDA280: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v288)
,	/* CT_v289: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA280: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA280))
, bytes2word(1,0,0,1)
, useLabel(CT_v290)
,};
Node FN_Control_46Category_46_95_46id[] = {
  bytes2word(ZAP_ARG_I1,NEEDHEAP_I32,HEAP_CVAL_I3,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I4,HEAP_OFF_N1,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v290: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Control_46Category_46_95_46id[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_95_46id),1)
, VAPTAG(useLabel(FN_LAMBDA281))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95noMethodError))
, bytes2word(0,0,0,0)
, useLabel(CT_v292)
,	/* FN_LAMBDA281: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v291)
,	/* CT_v292: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA281: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA281))
, bytes2word(1,0,0,1)
, useLabel(CT_v293)
,};
Node FN_Control_46Category_46_46[] = {
  bytes2word(NEEDSTACK_I16,SELECTOR_EVAL,SELECT,1)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v293: (byte 0) */
  HW(0,1)
, 0
,};
Node F0_Control_46Category_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_46),1)
, bytes2word(1,0,0,1)
, useLabel(CT_v294)
,};
Node FN_Control_46Category_46id[] = {
  bytes2word(NEEDSTACK_I16,SELECTOR_EVAL,SELECT,2)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v294: (byte 0) */
  HW(0,1)
, 0
,};
Node F0_Control_46Category_46id[] = {
  CAPTAG(useLabel(FN_Control_46Category_46id),1)
, bytes2word(0,0,0,0)
, useLabel(CT_v295)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_N1,1)
, bytes2word(HEAP_CVAL_I3,HEAP_CVAL_I4,RETURN,ENDCODE)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,2,0)
,	/* CT_v295: (byte 0) */
  HW(2,0)
, 0
,};
Node CF_Control_46Category_46Category_46Prelude_46_45_62[] = {
  VAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62))
, useLabel(F0_Control_46Category_46Category_46Prelude_46_45_62_46_46)
, useLabel(CF_Control_46Category_46Category_46Prelude_46_45_62_46id)
,	/* ST_v288: (byte 0) */
  bytes2word(78,111,32,100)
, bytes2word(101,102,97,117)
, bytes2word(108,116,32,100)
, bytes2word(101,102,105,110)
, bytes2word(105,116,105,111)
, bytes2word(110,32,102,111)
, bytes2word(114,32,99,108)
, bytes2word(97,115,115,32)
, bytes2word(109,101,116,104)
, bytes2word(111,100,32,46)
,	/* ST_v291: (byte 1) */
  bytes2word(0,78,111,32)
, bytes2word(100,101,102,97)
, bytes2word(117,108,116,32)
, bytes2word(100,101,102,105)
, bytes2word(110,105,116,105)
, bytes2word(111,110,32,102)
, bytes2word(111,114,32,99)
, bytes2word(108,97,115,115)
, bytes2word(32,109,101,116)
, bytes2word(104,111,100,32)
, bytes2word(105,100,0,0)
,};
