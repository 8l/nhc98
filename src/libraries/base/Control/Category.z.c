#include "newmacros.h"
#include "runtime.h"

#define CT_v284	((void*)startLabel+44)
#define CT_v286	((void*)startLabel+92)
#define CT_v288	((void*)startLabel+144)
#define CT_v290	((void*)startLabel+192)
#define CT_v292	((void*)startLabel+256)
#define FN_LAMBDA280	((void*)startLabel+284)
#define CT_v295	((void*)startLabel+316)
#define CF_LAMBDA280	((void*)startLabel+324)
#define CT_v297	((void*)startLabel+364)
#define FN_LAMBDA281	((void*)startLabel+392)
#define CT_v300	((void*)startLabel+424)
#define CF_LAMBDA281	((void*)startLabel+432)
#define CT_v302	((void*)startLabel+468)
#define CT_v304	((void*)startLabel+512)
#define CT_v306	((void*)startLabel+564)
#define ST_v282	((void*)startLabel+584)
#define ST_v301	((void*)startLabel+601)
#define ST_v285	((void*)startLabel+620)
#define ST_v283	((void*)startLabel+641)
#define ST_v305	((void*)startLabel+662)
#define ST_v287	((void*)startLabel+699)
#define ST_v289	((void*)startLabel+738)
#define ST_v291	((void*)startLabel+778)
#define ST_v293	((void*)startLabel+799)
#define ST_v296	((void*)startLabel+826)
#define ST_v298	((void*)startLabel+848)
#define ST_v303	((void*)startLabel+876)
#define ST_v294	((void*)startLabel+896)
#define ST_v299	((void*)startLabel+937)
extern Node TM_Control_46Category[];
extern Node FN_Control_46Category_46_46[];
extern Node FN_Prelude_46_46[];
extern Node F0_Prelude_46id[];
extern Node FN_NHC_46Internal_46_95noMethodError[];
extern Node TMSUB_Control_46Category[];

static Node startLabel[] = {
  bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v284)
,};
Node FN_Control_46Category_46_62_62_62[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_ARG_I2,PUSH_ARG_I3,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I3,HEAP_ARG,1,ZAP_ARG_I1)
, bytes2word(ZAP_ARG_I2,ZAP_ARG_I3,EVAL,NEEDHEAP_I32)
, bytes2word(APPLY,2,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
, 510001
, useLabel(ST_v283)
,	/* CT_v284: (byte 0) */
  HW(1,3)
, 0
,};
Node F0_Control_46Category_46_62_62_62[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_62_62_62),3)
, VAPTAG(useLabel(FN_Control_46Category_46_46))
, bytes2word(1,0,0,1)
, useLabel(CT_v286)
,};
Node FN_Control_46Category_46_60_60_60[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_ARG)
, bytes2word(1,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
, 470002
, useLabel(ST_v285)
,	/* CT_v286: (byte 0) */
  HW(1,1)
, 0
,};
Node F0_Control_46Category_46_60_60_60[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_60_60_60),1)
, VAPTAG(useLabel(FN_Control_46Category_46_46))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v288)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62_46_46[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_ARG_ARG_RET_EVAL)
, bytes2word(1,2,ENDCODE,0)
, bytes2word(0,0,0,0)
, 420010
, useLabel(ST_v287)
,	/* CT_v288: (byte 0) */
  HW(1,2)
, 0
,};
Node F0_Control_46Category_46Category_46Prelude_46_45_62_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62_46_46),2)
, VAPTAG(useLabel(FN_Prelude_46_46))
, bytes2word(0,0,0,0)
, useLabel(CT_v290)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62_46id[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDSTACK_I16,PUSH_CVAL_P1,3,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 390009
, useLabel(ST_v289)
,	/* CT_v290: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_Control_46Category_46Category_46Prelude_46_45_62_46id[] = {
  VAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62_46id))
, useLabel(F0_Prelude_46id)
, bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v292)
,};
Node FN_Control_46Category_46_95_46_46[] = {
  useLabel(TM_Control_46Category)
, bytes2word(ZAP_ARG_I1,NEEDHEAP_I32,PUSH_ARG_I3,PUSH_ARG_I2)
, bytes2word(HEAP_CVAL_I3,PUSH_HEAP,HEAP_CVAL_I4,HEAP_OFF_N1)
, bytes2word(2,ZAP_ARG_I2,ZAP_ARG_I3,EVAL)
, bytes2word(NEEDHEAP_I32,APPLY,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 0
, useLabel(ST_v291)
,	/* CT_v292: (byte 0) */
  HW(2,3)
, 0
,};
Node F0_Control_46Category_46_95_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_95_46_46),3)
, VAPTAG(useLabel(FN_LAMBDA280))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95noMethodError))
, bytes2word(0,0,0,0)
, useLabel(CT_v295)
,	/* FN_LAMBDA280: (byte 0) */
  useLabel(TMSUB_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,4,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v294)
, 0
, useLabel(ST_v293)
,	/* CT_v295: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA280: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA280))
, bytes2word(1,0,0,1)
, useLabel(CT_v297)
,};
Node FN_Control_46Category_46_95_46id[] = {
  useLabel(TM_Control_46Category)
, bytes2word(ZAP_ARG_I1,NEEDHEAP_I32,HEAP_CVAL_I3,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I4,HEAP_OFF_N1,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 0
, useLabel(ST_v296)
,	/* CT_v297: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Control_46Category_46_95_46id[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_95_46id),1)
, VAPTAG(useLabel(FN_LAMBDA281))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95noMethodError))
, bytes2word(0,0,0,0)
, useLabel(CT_v300)
,	/* FN_LAMBDA281: (byte 0) */
  useLabel(TMSUB_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,4,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v299)
, 0
, useLabel(ST_v298)
,	/* CT_v300: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA281: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA281))
, bytes2word(1,0,0,1)
, useLabel(CT_v302)
,};
Node FN_Control_46Category_46_46[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDSTACK_I16,SELECTOR_EVAL,SELECT,1)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 220007
, useLabel(ST_v301)
,	/* CT_v302: (byte 0) */
  HW(0,1)
, 0
,};
Node F0_Control_46Category_46_46[] = {
  CAPTAG(useLabel(FN_Control_46Category_46_46),1)
, bytes2word(1,0,0,1)
, useLabel(CT_v304)
,};
Node FN_Control_46Category_46id[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDSTACK_I16,SELECTOR_EVAL,SELECT,2)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 220007
, useLabel(ST_v303)
,	/* CT_v304: (byte 0) */
  HW(0,1)
, 0
,};
Node F0_Control_46Category_46id[] = {
  CAPTAG(useLabel(FN_Control_46Category_46id),1)
, bytes2word(0,0,0,0)
, useLabel(CT_v306)
,};
Node FN_Control_46Category_46Category_46Prelude_46_45_62[] = {
  useLabel(TM_Control_46Category)
, bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_IN3,HEAP_CVAL_I3)
, bytes2word(HEAP_CVAL_I4,RETURN,ENDCODE,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,2,0)
, 380010
, useLabel(ST_v305)
,	/* CT_v306: (byte 0) */
  HW(2,0)
, 0
,};
Node CF_Control_46Category_46Category_46Prelude_46_45_62[] = {
  VAPTAG(useLabel(FN_Control_46Category_46Category_46Prelude_46_45_62))
, useLabel(F0_Control_46Category_46Category_46Prelude_46_45_62_46_46)
, useLabel(CF_Control_46Category_46Category_46Prelude_46_45_62_46id)
,};
Node PM_Control_46Category[] = {
 	/* ST_v282: (byte 0) */
  bytes2word(67,111,110,116)
, bytes2word(114,111,108,46)
, bytes2word(67,97,116,101)
, bytes2word(103,111,114,121)
,	/* ST_v301: (byte 1) */
  bytes2word(0,67,111,110)
, bytes2word(116,114,111,108)
, bytes2word(46,67,97,116)
, bytes2word(101,103,111,114)
,	/* ST_v285: (byte 4) */
  bytes2word(121,46,46,0)
, bytes2word(67,111,110,116)
, bytes2word(114,111,108,46)
, bytes2word(67,97,116,101)
, bytes2word(103,111,114,121)
, bytes2word(46,60,60,60)
,	/* ST_v283: (byte 1) */
  bytes2word(0,67,111,110)
, bytes2word(116,114,111,108)
, bytes2word(46,67,97,116)
, bytes2word(101,103,111,114)
, bytes2word(121,46,62,62)
,	/* ST_v305: (byte 2) */
  bytes2word(62,0,67,111)
, bytes2word(110,116,114,111)
, bytes2word(108,46,67,97)
, bytes2word(116,101,103,111)
, bytes2word(114,121,46,67)
, bytes2word(97,116,101,103)
, bytes2word(111,114,121,46)
, bytes2word(80,114,101,108)
, bytes2word(117,100,101,46)
,	/* ST_v287: (byte 3) */
  bytes2word(45,62,0,67)
, bytes2word(111,110,116,114)
, bytes2word(111,108,46,67)
, bytes2word(97,116,101,103)
, bytes2word(111,114,121,46)
, bytes2word(67,97,116,101)
, bytes2word(103,111,114,121)
, bytes2word(46,80,114,101)
, bytes2word(108,117,100,101)
, bytes2word(46,45,62,46)
,	/* ST_v289: (byte 2) */
  bytes2word(46,0,67,111)
, bytes2word(110,116,114,111)
, bytes2word(108,46,67,97)
, bytes2word(116,101,103,111)
, bytes2word(114,121,46,67)
, bytes2word(97,116,101,103)
, bytes2word(111,114,121,46)
, bytes2word(80,114,101,108)
, bytes2word(117,100,101,46)
, bytes2word(45,62,46,105)
,	/* ST_v291: (byte 2) */
  bytes2word(100,0,67,111)
, bytes2word(110,116,114,111)
, bytes2word(108,46,67,97)
, bytes2word(116,101,103,111)
, bytes2word(114,121,46,95)
,	/* ST_v293: (byte 3) */
  bytes2word(46,46,0,67)
, bytes2word(111,110,116,114)
, bytes2word(111,108,46,67)
, bytes2word(97,116,101,103)
, bytes2word(111,114,121,46)
, bytes2word(95,46,46,58)
, bytes2word(110,111,112,111)
,	/* ST_v296: (byte 2) */
  bytes2word(115,0,67,111)
, bytes2word(110,116,114,111)
, bytes2word(108,46,67,97)
, bytes2word(116,101,103,111)
, bytes2word(114,121,46,95)
,	/* ST_v298: (byte 4) */
  bytes2word(46,105,100,0)
, bytes2word(67,111,110,116)
, bytes2word(114,111,108,46)
, bytes2word(67,97,116,101)
, bytes2word(103,111,114,121)
, bytes2word(46,95,46,105)
, bytes2word(100,58,110,111)
,	/* ST_v303: (byte 4) */
  bytes2word(112,111,115,0)
, bytes2word(67,111,110,116)
, bytes2word(114,111,108,46)
, bytes2word(67,97,116,101)
, bytes2word(103,111,114,121)
,	/* ST_v294: (byte 4) */
  bytes2word(46,105,100,0)
, bytes2word(78,111,32,100)
, bytes2word(101,102,97,117)
, bytes2word(108,116,32,100)
, bytes2word(101,102,105,110)
, bytes2word(105,116,105,111)
, bytes2word(110,32,102,111)
, bytes2word(114,32,99,108)
, bytes2word(97,115,115,32)
, bytes2word(109,101,116,104)
, bytes2word(111,100,32,46)
,	/* ST_v299: (byte 1) */
  bytes2word(0,78,111,32)
, bytes2word(100,101,102,97)
, bytes2word(117,108,116,32)
, bytes2word(100,101,102,105)
, bytes2word(110,105,116,105)
, bytes2word(111,110,32,102)
, bytes2word(111,114,32,99)
, bytes2word(108,97,115,115)
, bytes2word(32,109,101,116)
, bytes2word(104,111,100,32)
, bytes2word(105,100,0,0)
,};
