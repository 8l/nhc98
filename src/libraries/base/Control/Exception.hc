#include "newmacros.h"
#include "runtime.h"

#define FN_Control_46Exception_46catchesHandler	((void*)startLabel+16)
#define CT_v261	((void*)startLabel+44)
#define F0_Control_46Exception_46catchesHandler	((void*)startLabel+52)
#define FN_Control_46Exception_46Prelude_46214_46tryHandler	((void*)startLabel+88)
#define v265	((void*)startLabel+110)
#define v266	((void*)startLabel+113)
#define CT_v269	((void*)startLabel+136)
#define F0_Control_46Exception_46Prelude_46214_46tryHandler	((void*)startLabel+144)
#define CT_v270	((void*)startLabel+184)
extern Node FN_Control_46Exception_46Base_46throw[];
extern Node CF_Control_46Exception_46Base_46Exception_46Control_46Exception_46Base_46SomeException[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46foldr[];
extern Node FN_Control_46Exception_46Base_46fromException[];
extern Node FN_Control_46Exception_46Base_46catch[];

static Node startLabel[] = {
  42
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v261)
,	/* FN_Control_46Exception_46catchesHandler: (byte 0) */
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,2)
, bytes2word(HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_CVAL_P1,6)
, bytes2word(HEAP_OFF_N1,3,HEAP_ARG,2)
, bytes2word(PUSH_HEAP,HEAP_CVAL_P1,7,HEAP_OFF_N1)
, bytes2word(8,HEAP_OFF_N1,5,HEAP_ARG)
, bytes2word(1,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v261: (byte 0) */
  HW(5,2)
, 0
,	/* F0_Control_46Exception_46catchesHandler: (byte 0) */
  CAPTAG(useLabel(FN_Control_46Exception_46catchesHandler),2)
, CAPTAG(useLabel(FN_Control_46Exception_46Prelude_46214_46tryHandler),2)
, VAPTAG(useLabel(FN_Control_46Exception_46Base_46throw))
, useLabel(CF_Control_46Exception_46Base_46Exception_46Control_46Exception_46Base_46SomeException)
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46foldr))
, bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v269)
,	/* FN_Control_46Exception_46Prelude_46214_46tryHandler: (byte 0) */
  bytes2word(NEEDSTACK_I16,PUSH_ZAP_ARG_I2,EVAL,NEEDHEAP_I32)
, bytes2word(UNPACK,2,PUSH_ARG_I1,PUSH_HEAP)
, bytes2word(HEAP_CVAL_I3,HEAP_I2,ZAP_ARG_I1,EVAL)
, bytes2word(NEEDHEAP_I32,APPLY,1,EVAL)
, bytes2word(TABLESWITCH,2,TOP(4),BOT(4))
,	/* v265: (byte 2) */
  bytes2word(TOP(7),BOT(7),POP_I1,PUSH_ARG_I3)
,	/* v266: (byte 1) */
  bytes2word(RETURN_EVAL,UNPACK,1,PUSH_P1)
, bytes2word(0,PUSH_P1,3,ZAP_ARG_I3)
, bytes2word(ZAP_STACK_P1,4,ZAP_STACK_P1,3)
, bytes2word(ZAP_STACK_P1,2,EVAL,NEEDHEAP_I32)
, bytes2word(APPLY,1,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v269: (byte 0) */
  HW(1,3)
, 0
,	/* F0_Control_46Exception_46Prelude_46214_46tryHandler: (byte 0) */
  CAPTAG(useLabel(FN_Control_46Exception_46Prelude_46214_46tryHandler),3)
, VAPTAG(useLabel(FN_Control_46Exception_46Base_46fromException))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v270)
,};
Node FN_Control_46Exception_46catches[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,2)
, bytes2word(PUSH_HEAP,HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_ARG)
, bytes2word(1,HEAP_OFF_N1,5,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v270: (byte 0) */
  HW(3,2)
, 0
,};
Node F0_Control_46Exception_46catches[] = {
  CAPTAG(useLabel(FN_Control_46Exception_46catches),2)
, CAPTAG(useLabel(FN_Control_46Exception_46catchesHandler),1)
, VAPTAG(useLabel(FN_Control_46Exception_46Base_46catch))
, useLabel(CF_Control_46Exception_46Base_46Exception_46Control_46Exception_46Base_46SomeException)
,};
