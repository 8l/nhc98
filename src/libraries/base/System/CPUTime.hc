#include "newmacros.h"
#include "runtime.h"

#define FN_System_46CPUTime_46realToInteger	((void*)startLabel+12)
#define CT_v186	((void*)startLabel+36)
#define F0_System_46CPUTime_46realToInteger	((void*)startLabel+44)
extern Node FN_Prelude_46realToFrac[];
extern Node CF_Prelude_46Fractional_46Prelude_46Double[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46RealFrac_46Prelude_46Double_46round[];
extern Node CF_Prelude_46Integral_46Prelude_46Integer[];

static Node startLabel[] = {
  bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v186)
,	/* FN_System_46CPUTime_46realToInteger: (byte 0) */
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,1)
, bytes2word(HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_OFF_N1,4)
, bytes2word(HEAP_ARG,2,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(6,HEAP_CVAL_P1,7,HEAP_OFF_N1)
, bytes2word(5,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v186: (byte 0) */
  HW(5,2)
, 0
,	/* F0_System_46CPUTime_46realToInteger: (byte 0) */
  CAPTAG(useLabel(FN_System_46CPUTime_46realToInteger),2)
, VAPTAG(useLabel(FN_Prelude_46realToFrac))
, useLabel(CF_Prelude_46Fractional_46Prelude_46Double)
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46RealFrac_46Prelude_46Double_46round))
, useLabel(CF_Prelude_46Integral_46Prelude_46Integer)
,};
