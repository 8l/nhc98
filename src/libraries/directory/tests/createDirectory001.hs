import System.Directory
import Control.Exception

testdir = "createDirectory001.dir"

main = do
  try (removeDirectory testdir) :: IO (Either IOException ())
  createDirectory testdir
  r <- try $ createDirectory testdir
  print (r :: Either IOException ()) -- already exists
  removeDirectory testdir

