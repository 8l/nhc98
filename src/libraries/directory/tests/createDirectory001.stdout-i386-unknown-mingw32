Left CreateDirectory "createDirectory001.dir": already exists (Cannot create a file when that file already exists.)
