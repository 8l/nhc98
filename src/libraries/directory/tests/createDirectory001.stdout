Left createDirectory001.dir: createDirectory: already exists (File exists)
